### PR TITLE
Implement unit selection

### PR DESCRIPTION
**Pull request by achilleasa**
This PR brings time unit selection for the `print` and `diff` commands. Users can now pass the `display-unit` option to force a specific time unit (one of `ms`, `us` or `ns`) or to let prism auto-detect the proper unit by specifying `auto`.

In addition, this PR simplifies the diff command output and updates the thresholding code to also work with percentages (see https://github.com/geckoboard/prism/commit/1531bd6bbc706f3847a2543027531229dac0d8d3)

### CODE REVIEW

File: cmd/diff.go [hunk 1]
```diff
@@ -195,18 +214,64 @@ func populateDiffRow(rootMetrics *profiler.CallMetrics, correlation *correlatedM
...
+				case tableColP90:
+					val = metrics.P90Time
+				case tableColP99:
+					val = metrics.P99Time
+				default:
+					continue
+				}
+
+				dUnit := detectTimeUnit(val)
+				if dUnit > unit {
```
Comment (achilleasa) [inline cmd/diff.go]: time units are sorted so ms > us > ns; this test checks whether we need more accuracy than what we have detected this far

File: cmd/diff.go [hunk 2]
```diff
@@ -257,27 +322,32 @@ func fmtDiff(root, baseLine, candidate *profiler.CallMetrics, metricType tableCo
...
+	baseTime := dp.unit.Convert(baseVal)
+	candTime := dp.unit.Convert(candVal)
 
-	if candidate == baseLine || absDelta < threshold {
-		switch df {
+	if candidate == baseLine {
+		switch dp.format {
 		case displayTime:
-			return fmt.Sprintf("%1.2f (--)", candMs)
+			return fmt.Sprintf("%s", dp.unit.Format(candTime))
```
Comment (leocassarani) [inline cmd/diff.go]: Why do we need the `%s` here – could it not just be `return dp.unit.Format(candTime)` (or maybe `.String()`)?

File: cmd/diff.go [hunk 3]
```diff
@@ -192,14 +198,59 @@ func (dp *diffPrinter) Tabularize(profiles []*profiler.Profile, correlations []*
...
+	// For each column insert enough whitespace before the parenthesized
+	// block so that the numbers are aligned correctly
+	for rowIndex, row := range dp.rows {
+		for colIndex, col := range row {
+			parenIndex := strings.LastIndexByte(col, '(')
+			if parenIndex == -1 {
+				continue
+			}
+
+			padding := maxComparisonLen[colIndex] - (len(col) - parenIndex - 1)
```
Comment (achilleasa) [inline cmd/diff.go]: I think I may need to strip the ansi chars here too; let me double-check that

**Comment by leocassarani** [comment]
:cake: This is a nice addition! :smile:

**Pull request merged**